spec/Section 6 -- Execution.md

```diff
@@ -287,17 +294,33 @@ operation type.
 
 #### Response Stream
 
-Each event in the underlying Source Stream triggers execution of the
-subscription _selection set_ using that event as a root value.
+Each event from the underlying Source Stream triggers execution of the
+subscription _selection set_ using that event's value as the {initialValue}.
 
 MapSourceToResponseEvent(sourceStream, subscription, schema, variableValues):
 
-- Return a new event stream {responseStream} which yields events as follows:
-  - For each {event} on {sourceStream}:
-    - Let {response} be the result of running
-      {ExecuteSubscriptionEvent(subscription, schema, variableValues, event)}.
-    - Yield an event containing {response}.
-  - When {sourceStream} completes: complete {responseStream}.
+- Let {responseStream} be a new _event stream_.
+- When {sourceStream} emits {sourceValue}:
+  - Let {response} be the result of running
+    {ExecuteSubscriptionEvent(subscription, schema, variableValues,
+    sourceValue)}.
+  - If internal {error} was raised:
+    - Cancel {sourceStream}.
+    - Complete {responseStream} with {error}.
+  - Otherwise emit {response} on {responseStream}.
+- When {sourceStream} completes normally:
+  - Complete {responseStream} normally.
+- When {sourceStream} completes with {error}:
+  - Complete {responseStream} with {error}.
```

Comment:
I agree that for normal errors in `sourceStream` we should capture them and emit an `{ errors: [ ... ] }` payload rather than terminating with an error.

```suggestion
  - Let {errors} be a list containing {error}.
  - Let {response} be an unordered map containing {errors}.
  - Emit {response} on {responseStream}.
  - Complete {responseStream} normally.
```

However, this is a change in behavior (not an editorial change like the rest of this PR), so perhaps this should be raised _after_ this editorial only change is merged - see https://github.com/graphql/graphql-spec/pull/1127 for a suitable follow-up PR.